scripts/test-cve-2016-7054.py

```diff
@@ -0,0 +1,151 @@
+# Authors: Babak Amin Azad, Milan Lysonek
+"""Test for CVE-2016-7054 - ChaCha20/Poly1305 heap-buffer-overflow"""
+
+from __future__ import print_function
+import traceback
+import sys
+import getopt
+import re
+
+from tlsfuzzer.runner import Runner
+from tlsfuzzer.messages import Connect, ClientHelloGenerator, \
+        ClientKeyExchangeGenerator, ChangeCipherSpecGenerator, \
+        FinishedGenerator, ApplicationDataGenerator, \
+        fuzz_encrypted_message, AlertGenerator
+from tlsfuzzer.expect import ExpectServerHello, ExpectCertificate, \
+        ExpectServerHelloDone, ExpectChangeCipherSpec, ExpectFinished, \
+        ExpectAlert, ExpectClose, ExpectServerKeyExchange, \
+        ExpectApplicationData
+from tlslite.constants import CipherSuite, AlertLevel, AlertDescription
+
+
+def natural_sort_keys(s, _nsre=re.compile('([0-9]+)')):
+    return [int(text) if text.isdigit() else text.lower()
+            for text in re.split(_nsre, s)]
+
+
+def help_msg():
+    print("Usage: <script-name> [-h hostname] [-p port] [[probe-name] ...]")
+    print(" -h hostname    name of the host to run the test against")
+    print("                localhost by default")
+    print(" -p port        port number to use for connection, 4433 by default")
+    print(" probe-name     if present, will run only the probes with given")
+    print("                names and not all of them, e.g \"sanity\"")
+    print(" -e probe-name  exclude the probe from the list of the ones run")
+    print("                may be specified multiple times")
+    print(" --help         this message")
+
+
+def main():
+    host = "localhost"
+    port = 4433
+    run_exclude = set()
+
+    argv = sys.argv[1:]
+    opts, args = getopt.getopt(argv, "h:p:e:", ["help"])
+    for opt, arg in opts:
+        if opt == '-h':
+            host = arg
+        elif opt == '-p':
+            port = int(arg)
+        elif opt == '-e':
+            run_exclude.add(arg)
+        elif opt == '--help':
+            help_msg()
+            sys.exit(0)
+        else:
+            raise ValueError("Unknown option: {0}".format(opt))
+
+    if args:
+        run_only = set(args)
+    else:
+        run_only = None
+
+    conversations = {}
+
+    # 16 chars: POLY1305 tag 128 bit
+    # Tampering one bit suffices to damage the mac
+    # The payload has to be long enough to trigger heap overflow
+    n = 15000
```

Comment:
At CVE specification is:
> corrupting larger payloads
So I used same length as was length used at old PR. What length should I use? `2**14 - 24` as max. length - 24 bytes (GET request length + "X-test: ")?